pysmt/solvers/bdd.py

```diff
@@ -54,13 +55,56 @@ def __enter__(self):
         LockDdManager._depth += 1
 
     def __exit__(self, type, value, traceback):
-        if not pycudd.GetDefaultDdManager() == self.manager:
-            warnings.warn("The default DdManager changed without a " \
-                          "context protecting it")
-        LockDdManager._depth -= 1
-        if LockDdManager._depth == 0:
-            pycudd.ResetDefaultDdManager()
-
+        try:
+            assert self.manager is not None
+            current_manager = pycudd.GetDefaultDdManager()
+            if not current_manager == self.manager:
+                warnings.warn("The default DdManager changed without a " \
+                              "context protecting it", stacklevel=2)
+        finally:
+            LockDdManager._depth -= 1
+            if LockDdManager._depth == 0:
+                pycudd.ResetDefaultDdManager()
+
+
+
+class BddOptions(SolverOptions):
+
+    ## CUDD Reordering algorithms
+    CUDD_REORDER_SAME, \
+        CUDD_REORDER_NONE, \
+        CUDD_REORDER_RANDOM, \
+        CUDD_REORDER_RANDOM_PIVOT, \
+        CUDD_REORDER_SIFT, \
+        CUDD_REORDER_SIFT_CONVERGE, \
+        CUDD_REORDER_SYMM_SIFT, \
+        CUDD_REORDER_SYMM_SIFT_CONV, \
+        CUDD_REORDER_WINDOW2, \
+        CUDD_REORDER_WINDOW3, \
+        CUDD_REORDER_WINDOW4, \
+        CUDD_REORDER_WINDOW2_CONV, \
+        CUDD_REORDER_WINDOW3_CONV, \
+        CUDD_REORDER_WINDOW4_CONV, \
+        CUDD_REORDER_GROUP_SIFT, \
+        CUDD_REORDER_GROUP_SIFT_CONV, \
+        CUDD_REORDER_ANNEALING, \
+        CUDD_REORDER_GENETIC, \
+        CUDD_REORDER_LINEAR, \
+        CUDD_REORDER_LINEAR_CONVERGE, \
+        CUDD_REORDER_LAZY_SIFT, \
+        CUDD_REORDER_EXACT = range(22)
```

Comment:
no, seems like no. This is a bit fragile, but until we have a better wrapper for cudd, this is ok.